### PR TITLE
Fix PyPI page

### PR DESCRIPTION
**Pull request by jimfulton**
I noticed the other day that the PyPI page for the current release is hosed WRT ReST:

https://pypi.python.org/pypi/ZODB/5.2.0

In fixing this, I realized that a lot of text was out of date and not well suited to a PyPI page, so I:

- Split developer info into a separate doc.

- Simplified the README and setup script.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -5,14 +5,15 @@
 5.2.1 (unreleased)
 ==================
 
-- Nothing changed yet.
-
+- Fixed: When opening FileStorages in read-only mode, non-existent
+  files were silently created.  Creating a file-storage against a
```
Comment (jamadden) [inline CHANGES.rst]: "Creating a read-only FileStorage..."

Leaving out the 'read-only' makes that sentence sound a little scary.

File: setup.py [hunk 2]
```diff
@@ -100,19 +91,11 @@ def emit(self, record):
...
-    with open(os.path.join(*file_path), 'rb') as file:
-        return file.read()
-
-long_description = str(
-    ("\n".join(doclines[2:]) + "\n\n" +
-     ".. contents::\n\n" +
-     read_file("README.rst").decode('latin-1')  + "\n\n" +
-     read_file("CHANGES.rst").decode('latin-1')))
+long_description = (
+    open("README.rst").read()  +
```
Comment (jamadden) [inline setup.py]: Please bring back `read_file` so that the files get closed and not leaked on PyPy and Python 3.

File: setup.py [hunk 3]
```diff
@@ -128,7 +111,7 @@ def read_file(*path):
       url = 'http://www.zodb.org/',
       license = "ZPL 2.1",
       platforms = ["any"],
-      description = doclines[0],
+      description = long_description[1],
```
Comment (jamadden) [inline setup.py]:
```pycon
>>> long_description = (
...     open("README.rst").read()  +
...    "\n\n" +
...     open("CHANGES.rst").read()
...  )
>>> long_description[1]
'='
```

Doesn't seem right.

File: DEVELOPERS.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
+For developers of ZODB
```
Comment (jamadden) [inline DEVELOPERS.rst]: This is no longer published anywhere on the web. Perhaps it could be referenced from a file included in doc/index.rst (.. include: ../DEVELOPERS.rst), and then it might be a good idea to add a link from the PyPI page to this info.

I'd also suggest adding an explicit link to the github repo *somewhere*, probably in this document. If I need to fix a bug (open a PR) in an unfamiliar package, I need to find its repo. First I check the PyPI page, it's often listed as the URL. But if not, I look for a "developing" page in the documentation which should list the repo. Bonus points if that just takes me one click from the PyPI page.

File: DEVELOPERS.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+======================
```
Comment (jamadden) [inline DEVELOPERS.rst]: Would you like to use the "standard" RST headers now that this is its own document? E.g., line above and below the document title? If you're in Emacs, C-c C-a C-s does that for you.

File: setup.py (hunk 2 above)
Comment (jimfulton) [inline setup.py]: This script is short lived. Why does that matter?

File: setup.py (hunk 3 above)
Comment (jimfulton) [inline setup.py]: oops

File: setup.py (hunk 2 above)
Comment (jamadden) [inline setup.py]: Because it's just good practice to clean up after yourself?

There are tools like pyroma and others that import the setup file and then do a bunch of analysis so the script is not necessarily short lived.

File: DEVELOPERS.rst (hunk 4 above)
Comment (jimfulton) [inline DEVELOPERS.rst]: Well, it's published here.

+1 to adding a repo URL to the readme and making it more explicit in the docs.

Eventually, I think development should be a topic in the docs.

This PR just aims to unscrew some things.

At a meta level, If my goal was to be perfect, I would probably have put this off and the PyPI page would have remained broken longer.

File: DEVELOPERS.rst (hunk 5 above)
Comment (jimfulton) [inline DEVELOPERS.rst]: Um, sure, whatever. :)

File: setup.py (hunk 2 above)
Comment (jimfulton) [inline setup.py]: Pyroma rns for, like, 2 seconds.

I'll add it back, but this just clutters the code. I object to cluttered code more than I object to a non-existent memory-management problem.

Comment (jamadden) [inline setup.py]: That's probably an artifact of our backgrounds, then, because I object to bare `open`s and relying on the GC to close resources...curse you Java :)

Comment (jimfulton) [inline setup.py]: There is no leak here.  If this script was long lived, the files would eventually be closed by the garbage collector. Even if this was long lived, which it isn't, there is only a potential problem if the program were opening files faster than the GC could reap them.

The pedantic is the enemy of the productive.

File: CHANGES.rst (hunk 1 above)
Comment (jimfulton) [inline CHANGES.rst]: fixed. Thanks.

File: setup.py (hunk 2 above)
Comment (jimfulton) [inline setup.py]: done

File: DEVELOPERS.rst (hunk 4 above)
Comment (jimfulton) [inline DEVELOPERS.rst]: I added links to the README. I chose to make a drive-by change to the docs.

File: DEVELOPERS.rst (hunk 5 above)
Comment (jimfulton) [inline DEVELOPERS.rst]: done

File: README.rst [hunk 6]
```diff
@@ -19,3 +19,8 @@ high-degree of transparency.
 ZODB is an ACID Transactional database.
 
 To learn more, visit: http://www.zodb.org
+
+The githib repository is: at https://github.com/zopefoundation/zodb
```
Comment (jamadden) [inline README.rst]: github -> github

File: setup.py (hunk 3 above)
Comment (jimfulton) [inline setup.py]: fixed

File: README.rst [hunk 7]
```diff
@@ -19,3 +19,8 @@ high-degree of transparency.
...
+
+If you're interested in contributing to ZODB itself, see the
+`developer notes <DEVELOPERS.rst>`_.
```
Comment (jamadden) [inline README.rst]: Although that renders just fine locally in `restview --long-description`, is that going to be a broken link on PyPI? I didn't know it would automatically render more than long-description.

File: setup.py (hunk 2 above)
Comment (jamadden) [inline setup.py]: Thank you. 

Now I'll sleep easier knowing that PyPy won't complain about this. After all, it's not like there's anything else worrying going on in the world, right? 😄

File: README.rst (hunk 6 above)
Comment (jimfulton) [inline README.rst]: Oops, fixed

File: README.rst (hunk 7 above)
Comment (jimfulton) [inline README.rst]: Gaaa, was thinking in terms of github, not pypi. Fixed.

File: DEVELOPERS.rst (hunk 4 above)
Comment (jamadden) [inline DEVELOPERS.rst]: That works for me!

**Comment by jamadden** [comment]
Thanks for this. I see that I am, and apologize for being, the one that originally introduced the typo that broke the PyPI rendering.

**Comment by jimfulton** [comment]
No worries. It's fixed now.  I wonder if we can start hiding old releases now.
